pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -608,15 +612,22 @@ func validateExecutionStatusVariablesExpressions(expressions []string, ptNames s
 			}
 			// check if it contains context variable accessing execution status - $(tasks.taskname.status)
 			if containsExecutionStatusRef(expression) {
-				// strip tasks. and .status from tasks.taskname.status to further verify task name
-				pt := strings.TrimSuffix(strings.TrimPrefix(expression, "tasks."), ".status")
+				var pt string
+				if strings.HasSuffix(expression, ".status") {
+					// strip tasks. and .status from tasks.taskname.status to further verify task name
+					pt = strings.TrimSuffix(strings.TrimPrefix(expression, "tasks."), ".status")
+				} else {
```

Comment:
NIT: checking for `strings.HasSuffix(expression, ".reason")` would be more robust here